src/compas_viewer/scene/lineobject.py

```diff
@@ -14,24 +17,21 @@ class LineObject(ViewerSceneObject, GeometryObject):
     """
 
     def __init__(self, line: Line, **kwargs):
-        if kwargs["show_lines"] is None:
-            kwargs["show_lines"] = True
         super().__init__(geometry=line, **kwargs)
+        self.geometry: Line
+        self.show_lines = True
+
+    @property
+    def points(self) -> Optional[list[Point]]:
+        """The points to be shown in the viewer."""
+        return [self.geometry.start, self.geometry.end]
+
+    @property
+    def lines(self) -> Optional[list[Line]]:
+        """The lines to be shown in the viewer."""
+        return [self.geometry]
 
-    def _read_points_data(self) -> DataType:
-        positions = [self.geometry.start, self.geometry.end]
-        colors = [
-            self.pointscolor.get(0, self.pointscolor["_default"]),  # type: ignore
-            self.pointscolor.get(1, self.pointscolor["_default"]),  # type: ignore
-        ]
-        elements = [[0], [1]]
-        return positions, colors, elements
-
-    def _read_lines_data(self) -> DataType:
-        positions = [self.geometry.start, self.geometry.end]
-        colors = [
-            self.linescolor.get(0, self.linescolor["_default"]),  # type: ignore
-            self.linescolor.get(1, self.linescolor["_default"]),  # type: ignore
-        ]
-        elements = [[0, 1]]
-        return positions, colors, elements
+    @property
+    def surfaces(self) -> Optional[list[Tuple[Point, Point, Point]]]:
+        """The surface to be shown in the viewer. Currently only triangles are supported."""
+        pass
```

Comment:
if the behaviour is based on an implicit `None` return, i would explicitly return `None`, instead of passing...